lib/index.js

```diff
@@ -92,7 +103,10 @@ module.exports = config => {
 
     getNextStep(req, res, callback) {
       const step = super.getNextStep(req, res, callback);
-      if (req.query.step === 'postcode') {
+      const addressFields = getFields(addressKey, req.translate, changeValidate(optional));
+      const addressRequired = _.find(addressFields[`${addressKey}-postcode`].validate, 'required');
+
+      if (req.query.step === 'postcode' && req.form.values[`${addressKey}-postcode`] && !addressRequired) {
```

Comment:
To clarify, you already have an `optional` flag, which you use on L106 to create a validator array, then use that validator array to determine if the address is required, but you know if the address is required because you have the `optional` flag.